core/services/request/files/FilesDataHandler.php

```diff
@@ -0,0 +1,237 @@
+     * @throws UnexpectedValueException
+     * @throws InvalidArgumentException
+     */
+    public function getFileObjects()
+    {
+        if (!$this->initialized) {
+            $this->initialize();
+        }
+        return $this->file_objects;
+    }
+
+    /**
+     * Sets up the file objects from the request's $_FILES data.
+     * @since $VID:$
+     * @throws UnexpectedValueException
+     * @throws InvalidArgumentException
+     */
+    protected function initialize()
+    {
+        $files_raw_data = $this->request->filesParams();
+        if (empty($files_raw_data)) {
+            return;
+        }
+        if ($this->isStrangeFilesArray($files_raw_data)) {
+            $data = $this->fixFilesDataArray($files_raw_data);
+        } else {
+            $data = $files_raw_data;
+        }
+        $this->file_objects = $this->createFileObjects($data);
+        $this->initialized = true;
+    }
+
+    /**
+     * Detects if $_FILES is a weird multi-dimensional array that needs fixing or not.
+     * @since $VID:$
+     * @param $files_data
+     * @return bool
+     * @throws UnexpectedValueException
+     */
+    protected function isStrangeFilesArray($files_data)
+    {
+        if (!is_array($files_data)) {
+            throw new UnexpectedValueException(
+                sprintf(
+                    esc_html__(
+                        'Unexpected PHP $_FILES data format. "%1$s" was expected to be an array.',
+                        'event_espresso'
+                    ),
+                    (string) $files_data
+                )
+            );
+        }
+        $first_value = reset($files_data);
+        if (!is_array($first_value)) {
+            throw new UnexpectedValueException(
+                sprintf(
+                    esc_html__(
+                        'Unexpected PHP $_FILES data format. "%1$s" was expected to be an array.',
+                        'event_espresso'
+                    ),
+                    (string) $first_value
+                )
+            );
+        }
+        $first_sub_array_item = reset($first_value);
+        if (is_array($first_sub_array_item)) {
+            // not just a 2d array
+            return true;
+        }
+        // yep, just 2d array
+        return false;
+    }
+
+    /**
+     * Takes into account that $_FILES does a weird thing when you have hierarchical form names (eg `<input type="file"
+     * name="my[hierarchical][form]">`): it leaves the top-level form part alone, but replaces the SECOND part with
+     * "name", "size", "tmp_name", etc. So that file's data is located at "my[name][hierarchical][form]",
+     * "my[size][hierarchical][form]", "my[tmp_name][hierarchical][form]", etc. It's really weird.
+     * @since $VID:$
+     * @param $files_data
+     * @return array
+     */
+    protected function fixFilesDataArray($files_data)
+    {
+        $sane_files_array = [];
+        foreach ($files_data as $top_key => $top_key_avlue) {
```

Comment:
missed this typo before: `$top_key_avlue` should be `$top_key_value`

Reply:
:bowing_man: